### PR TITLE
Fix corner-case bug for sgcr vs sgro

### PR DESCRIPTION
**Pull request by berland**
Uncovered by looking at untested code lines.

Several other tests added, and some fixed tests.

Merge, not squash.

### CODE REVIEW

File: pyscal/gasoil.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def __init__(
         self.sgcr = truncate_zeroness(sgcr, name="sgcr")
         self.sgro = truncate_zeroness(sgro, name="sgro")
 
-        if not (np.isclose(self.sgro, 0) or np.isclose(self.sgro, self.sgcr)):
+        if not (np.isclose(self.sgro, 0) or np.isclose(self.sgro - self.sgcr, 0)):
```
Comment (anders-kiaer) [inline pyscal/gasoil.py:105]: Out of interest: Why is `np.isclose(self.sgro - self.sgcr, 0)` necessary compared to `np.isclose(self.sgro, self.sgcr)`?

**Review by anders-kiaer — APPROVED** [review]
LGTM! A small question only.
